vslib/src/SwitchState.cpp

```diff
@@ -291,3 +293,55 @@ std::shared_ptr<saimeta::Meta> SwitchState::getMeta()
 
     return meta;
 }
+
+void SwitchState::findObjects(
+        _In_ sai_object_type_t object_type,
+        _In_ const sai_attribute_t &expect,
+        _Out_ std::vector<sai_object_id_t> &objects) const
+{
+    SWSS_LOG_ENTER();
+
+    objects.clear();
+    SaiAttrWrap expect_wrap(object_type, &expect);
+    const std::string expect_attr_name =
+        sai_serialize_attr_id(*expect_wrap.getAttrMetadata());
+
+    for (auto &obj : m_objectHash.at(object_type))
+    {
+        auto attr_itr = obj.second.find(expect_attr_name);
+
+        if (attr_itr != obj.second.end()
+            && attr_itr->second->getAttrStrValue() == expect_wrap.getAttrStrValue())
+        {
+            sai_object_id_t object_id = SAI_NULL_OBJECT_ID;
+            sai_deserialize_object_id(obj.first, object_id);
+            objects.push_back(object_id);
+        }
+
+    }
+
+}
+
+bool SwitchState::dumpObject(
```

Comment:
I want a function that can find out the attributes table according to object id, meanwhile I hope the type of attributes table is `std::vector<sai_attribute_t>`.
Because I have some functions as following, I use they to load MACsec attributes. I hope all of them can be used both in create, get, remove and set actions. 
The parameters of `create` and `set` actions include `sai_attribute_t *` which means I have opportunity to load MACsec Attr from their parameters. 
But parameters of `remove` and `get` actions only include object id, I have to find out those MACsec attributes  I needed according to the object ID. 
So I designed this function, `dumpObject`, to dump

https://github.com/Azure/sonic-sairedis/blob/d937d66426b3d62e10cb710dd192ad45b485ec1a/vslib/inc/SwitchStateBase.h#L473-L501